relayer/pkg/starknet/utils.go

```diff
@@ -77,6 +77,12 @@ func DecodeFelts(felts []*big.Int) ([]byte, error) {
 		bytesLen := Min(chunkSize, length)
 		bytesBuffer := buf[:bytesLen]
 
+		// TODO: this is inefficient because Bytes() and FillBytes() duplicate work
+		// reuse felt.Bytes()
+		if bytesLen < uint64(len(felt.Bytes())) {
+			return nil, errors.New("invalid: felt array can't be decoded")
+		}
```

Comment:
I tried dropping `FillBytes()` and using something like `data = append(data, bytes...)` but that fails some of the fuzz tests, need to use an intermediate buffer to copy into

Reply:
The issue here is in the fact that `0` are dropped when creating `big.Int`. For example:
```
data := []byte{0, 0, 0, 0, 0, 1, 2, 3}
felt := new(big.Int).SetBytes(data)
fmt.Println(len(felt.Bytes()))
```

Will output 3 while we assume that output shall be 8

Reply:
Since this is the case in 2nd point from issue there's no way differentiate
`felts[1] = new(big.Int).SetBytes([]byte{0x1, 0x2, 0x3, 0x4})`
from
`felts[1] = new(big.Int).SetBytes([]byte{0x0, 0x0, ..., 0x1, 0x2, 0x3, 0x4})`

So I think those changes are ok. Or the other way to resolve this all would be enforce EncodeFelts to take byte array with size is `n*31`,